src/Mod/Start/App/DisplayedFilesModel.cpp

```diff
@@ -38,30 +38,24 @@ using namespace Start;
 namespace
 {
 
-std::string humanReadableSize(unsigned int bytes)
+static std::string humanReadableSize(const size_t size)
 {
-    static const std::vector<std::string> siPrefix {
-        "b",
-        "kb",
-        "Mb",
-        "Gb",
-        "Tb",
-        "Pb",
-        "Eb"  // I think it's safe to stop here (for the time being)...
-    };
-    size_t base = 0;
-    double inUnits = bytes;
-    constexpr double siFactor {1000.0};
-    while (inUnits > siFactor && base < siPrefix.size() - 1) {
-        ++base;
-        inUnits /= siFactor;
-    }
-    if (base == 0) {
-        // Don't include a decimal point for bytes
-        return fmt::format("{:.0f} {}", inUnits, siPrefix[base]);
-    }
-    // For all others, include one digit after the decimal place
-    return fmt::format("{:.1f} {}", inUnits, siPrefix[base]);
+    static constexpr std::array<std::pair<char, unsigned long long>, 7> prefixes {
+        {{0, 0},
+         {'k', 1ULL << 10},
+         {'M', 1ULL << 20},
+         {'G', 1ULL << 30},
+         {'T', 1ULL << 40},
+         {'P', 1ULL << 50},
+         {'E', 1ULL << 60}}};
+
+    const auto spec = std::find_if(prefixes.rbegin(), prefixes.rend(), [&](const auto& pair) {
+        return pair.second <= size;
+    });

```

Comment:
Please use meaningful names. Pair is not meaningful, pair.second also not. If you insist on using the std::pair instead of proper struct please use name like `prefixMagnitudePair` and then structural binding to extract needed variables.

Reply:
Defining structure to hold two members and naming them just to give the reader clue which one is first and which one is second seems a bit overkill. Give it a try to see for yourself.
I'm not huge fan of giving local variables nice long descriptive names and I suck at it (descriptive naming in general), so I cannot really help here, sorry.

Reply:
It is not for which one is first and which one is second. This structure is not ordered, so being first or second does not provide any useful information.

Struct here is for providing what means what by providing well defined type. This is even hinted in the core guidelines: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f21-to-return-multiple-out-values-prefer-returning-a-struct. 

Cost of providing struct in code is almost none, you don't need to define it in a namespace or anything - `using` to define it in function will be absolutely fine. 

https://pedrorijo.com/blog/tuples-are-evil/.

Reply:
Look how you used that structure in your example bellow. You are still assigning `'k'` to the first member and  `1ULL << 10` to the second. That's why I wrote you immediately see which item is first and which second as it is written just above the place it is used.

Reply:
You only know that because of context. In case of array you have clear columns - one containing prefixes, one numbers. You can quickly understand the meaning without looking at the definition of the struct (well, except for the `{ 0, 0 }` case). 

In lambda you don't have such luxury and you have to consult code few lines above. Is it much? Nope. but when reading code you often don't read everything to the detail - only a party that is important for you. The logic of function is basically the find_if call and I'll start from there to understand what is going on - that is why having the name is important.

Reply:
Enforcing people reading the code to look for context and understand it is valuable for itself. Our opinions clearly differ here, fortunately this topic is not that important to me to continue arguing, so let's stick with `struct`.